### PR TITLE
Improved WebSocket messaging in await_deposit.rs and await_invoice.rs

### PR DESCRIPTION
**Pull request by OsauravO**
**This pull request makes the following changes:**
* Fixes issue #13
* I have made changes in handle_socket function  and accordingly in match_method in ws.rs file to improve the websocket messaging logic

### CODE REVIEW

**Comment by OsauravO** [comment]
@Kodylow  i have made some changes in handle_socket function  and accordingly in match_method in ws.rs file and also tried to explain my approach using comments . Accordingly i have changed handle_ws function in await_deposit.rs file . I am still getting some errors . Can you please check it for me . thanks

File: src/router/handlers/fedimint/ln/await_invoice.rs [hunk 1]
```diff
@@ -23,54 +24,37 @@ pub struct AwaitInvoiceRequest {
...
-                    anyhow!(reason),
-                ))
-            }
-            _ => {}
-        }
 
-        info!("Update: {update:?}");
+    while let Some(update) = updates.next().await {
+        let message = serde_json::to_string(&update).map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
+        ws_sender.send(Message::Text(message)).await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
```
Comment (Kodylow) [inline src/router/handlers/fedimint/ln/await_invoice.rs]: this is looking like the right direction but the request needs to be correlated to the response by the id, don't think that's being added in here. Also this is just putting text it should probably be json structured or use enums for the responses so they can be deterministically consumed by the client

File: src/router/handlers/fedimint/ln/await_invoice.rs [hunk 2]
```diff
@@ -23,54 +24,37 @@ pub struct AwaitInvoiceRequest {
...
-}
+// #[axum_macros::debug_handler]
+// pub async fn handle_rest(
+//     State(state): State<AppState>,
+//     Json(req): Json<AwaitInvoiceRequest>,
+// ) -> Result<Json<InfoResponse>, AppError> {
+//     let client = state.get_client(req.federation_id).await?;
+//     let invoice = _await_invoice(client, req).await?;
+//     Ok(Json(invoice))
+// }
```
Comment (Kodylow) [inline src/router/handlers/fedimint/ln/await_invoice.rs]: all the functionaliy needs to be able to handle both rest and websockets, you need to abstract it as such

File: src/router/handlers/fedimint/wallet/await_deposit.rs [hunk 3]
```diff
@@ -1,84 +1,63 @@
...
+    
+        let mut updates = client
+            .get_first_module::<WalletClientModule>()
+            .subscribe_deposit_updates(req.operation_id)
+            .await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?
+            .into_stream();
+    
+        while let Some(update) = updates.next().await {
+            let message = serde_json::to_string(&update).map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
+            ws_sender.send(Message::Text(message)).await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
```
Comment (Kodylow) [inline src/router/handlers/fedimint/wallet/await_deposit.rs]: again, enums and json structs > text messages. See how I structured the json responses for websockets previosuly, think it's in `ws.rs` under build response

**Review by Kodylow — CHANGES_REQUESTED** [review]

**Comment by OsauravO** [comment]
@Kodylow i tried to resolve all the changes you suggested but i am getting an error in ws.rs file . Can you please check it for me. Thanks

**Comment by Kodylow** [comment]
can you rebase + post the errors

**Comment by OsauravO** [comment]
i think i have done the rebasing now. 
and the error which i am getting in ws.rs file is
image

**Comment by Kodylow** [comment]
Okay, that error's pretty clear, you're passing ownership of the socket into the method which drops it so you have to either borrow it like `&socket`, clone it like `socket.clone()`, or refactor to not drop the socket. 

when you split the socket into sender/receiver you don't have the socket anymore, you have sender and receiver, so use those if you're splitting instead of using socket

**Comment by OsauravO** [comment]
hey i think i got what you said , can you now check it . Thanks a lot

**Comment by Kodylow** [comment]
Can you reopen this on https://github.com/fedimint/fedimint-clientd , moved the project into the fedimint org

**Comment by OsauravO** [comment]
Already done 👍

**Comment by Kodylow** [comment]
closing with move to new repo

**Pull request closed without merging**